### PR TITLE
Modularize sub_test.py

### PR DESCRIPTION
**Pull request by jhh67**
Make `sub_test.py `a module so that it can be imported by custom sub_test implementations.

### CODE REVIEW

File: util/test/sub_test.py [hunk 1]
```diff
@@ -740,1716 +741,1756 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
...
+    # Find the test util directory -- set this in start_test to permit
+    # a version of start_test other than the one in CHPL_HOME to be used
+    utildir=os.getenv('CHPL_TEST_UTIL_DIR');
+    if utildir is None or not os.path.isdir(utildir):
+        Fatal('Cannot find test util directory {0}'.format(utildir))
+
+    # Needed for MacOS mount points
+    utildir = os.path.realpath(utildir)
+    # sys.stdout.write('utildir='+utildir+'\n');
+
```
Comment (DanilaFe) [inline util/test/sub_test.py:826]: Could this also be factored into `get_utildir`?

File: util/test/sub_test.py [hunk 2]
```diff
@@ -740,1716 +741,1756 @@ def filter_errors(output_in, pre_exec_output, execgoodfile, execlog):
...
+    c_compiler = run_compileline('--compile', 'c compiler')
+    cpp_compiler = run_compileline('--compile-c++', 'c++ compiler')
+    host_c_compiler = run_compileline('--host-c-compiler', 'host c compiler')
+    host_cpp_compiler = run_compileline('--host-cxx-compiler', 'host c++ compiler')
+    runtime_includes_and_defines = run_compileline('--includes-and-defines',
+                                                   'runtime includes and defines')
+
+    # Find the test directory
+    testdir=chpl_home+'/test'
```
Comment (DanilaFe) [inline util/test/sub_test.py:849]: Is this code (and subsequent lines) duplicate now? It wasn't before -- could the `test_root_dir` computation be factored out?

**Review by DanilaFe — APPROVED** [review]
Git is displaying the diff in the most obnoxious manner possible; if your _real_ changes are only the functions at the top, and the rest is just indentation, this all looks good. I read up to line 977, and it was all indentation up to there.

It would be great if you ran a paratest with this to make sure everything still works.

File: util/test/sub_test.py (hunk 1 above)
Comment (jhh67) [reply util/test/sub_test.py:826]: Done

File: util/test/sub_test.py (hunk 2 above)
Comment (jhh67) [reply util/test/sub_test.py:849]: Good suggestion, I've done some additional refactoring.

**Pull request merged**